src/common/hooks.ts

```diff
@@ -45,15 +45,17 @@ export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
  * Redirects the user to their prior route, or home,
  * once authenticated.
  */
+
 export const useAuthenticatedRedirect = () => {
   const { isLoggedIn } = useSelector(selectSession);
 
-  const history = useHistory();
-  const location = useLocation<{ from?: string }>();
-
+  const navigate = useNavigate();
+  const location = useLocation();
   useEffect(() => {
     if (isLoggedIn) {
-      history.push(location.state?.from || "/home");
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      location?.state?.from || navigate("/home");
```

Comment:
So apparently in V6 we cannot add any types to useLocation. ( I was getting this error: `Expected 0 type arguments, but got 1`).

Reply:
Looks good, but